operator-pipeline-images/operatorcert/entrypoints/github_add_comment.py

```diff
@@ -0,0 +1,167 @@
+"""
+Script for adding a comment to a pull request or an issue.
+It can take either a filename or a comment as input and can
+post the comment back to GitHub accordingly.
+"""
+import argparse
+from ast import arg
+import logging
+import json
+import os
+import http.client
+import sys
+import urllib.parse
+
+from operatorcert import github, store_results
+from operatorcert.logger import setup_logger
+
+LOGGER = logging.getLogger("operator-cert")
+
+
+def setup_argparser() -> argparse.ArgumentParser:  # pragma: no cover
+    """
+    Setup argument parser
+
+    Returns:
+        Any: Initialized argument parser
+    """
+    parser = argparse.ArgumentParser(
+        description="Github cli tool to add comment to PR or an issue."
+    )
+    parser.add_argument(
+        "--github-host-url",
+        default="api.github.com",
+        help="The GitHub host, default: api.github.com",
+    )
+    parser.add_argument(
+        "--request-url",
+        required=True,
+        help="The GitHub issue or pull request URL where we want to add a new comment.",
+    )
+    parser.add_argument(
+        "--comment-or-file",
+        required=True,
+        help="The actual comment to add or the filename containing comment to post.",
+    )
+    parser.add_argument(
+        "--comment-tag", help="An invisible tag to be added into the comment."
+    )
+    parser.add_argument(
+        "--replace",
+        default="false",
+        help="When a tag is specified, and `REPLACE` is `true`, look for a comment with a matching tag and replace it with the new comment.",
+    )
+    parser.add_argument("--verbose", action="store_true", help="Verbose output")
+    return parser
+
+
+def github_add_comment(
+    github_host_url: str,
+    request_url: str,
+    comment_or_file: str,
+    commen_tag: str,
+    replace: str,
+) -> None:
+    split_url = urllib.parse.urlparse(request_url).path.split("/")
+
+    # This will convert https://github.com/foo/bar/pull/202 to
+    # api url path /repos/foo/issues/
+    api_url = "{base}/repos/{package}/issues/{id}".format(
```

Comment:
Please use f-strings instead of `format()`. https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python